src/Plugin/InstallTask/SiteSettings.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+namespace Drupal\stanford_profile\Plugin\InstallTask;
+
+use Drupal\Component\Utility\Html;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Site\Settings;
+use Drupal\externalauth\AuthmapInterface;
+use Drupal\stanford_profile\InstallTaskBase;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\GuzzleException;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * SNOW site settings installation.
+ *
+ * @InstallTask(
+ *   id="stanford_profile_site_settings"
+ * )
+ */
+class SiteSettings extends InstallTaskBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The fallback site name.
+   */
+  const DEFAULT_SITE = 'default';
+
+  /**
+   * Service now api endpoint.
+   */
+  const SNOW_API = 'https://stanford.service-now.com/api/stu/su_acsf_site_requester_information/requestor';
+
+  /**
+   * Guzzle service.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  protected $client;
+
+  /**
+   * Authmap service.
+   *
+   * @var \Drupal\externalauth\AuthmapInterface
+   */
+  protected $authmap;
+
+  /**
+   * Logger channel service.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelInterface
+   */
+  protected $logger;
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('http_client'),
+      $container->get('externalauth.authmap'),
+      $container->get('logger.factory')
+    );
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entityTypeManager, ClientInterface $client, AuthmapInterface $authmap, LoggerChannelFactoryInterface $logger_factory) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityTypeManager = $entityTypeManager;
+    $this->client = $client;
+    $this->authmap = $authmap;
+    $this->logger = $logger_factory->get('stanford_profile');
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function runTask(array &$install_state) {
+    // @codeCoverageIgnoreStart
+    if (!static::isAhEnv()) {
+      return;
+    }
+    // @codeCoverageIgnoreEnd
+    $site_name = $install_state['forms']['install_configure_form']['site_name'] ?? self::DEFAULT_SITE;
+    $site_name = Html::escape($site_name);
+
+    $site_data = $this->getSnowData($site_name);
+    if (empty($site_data)) {
+      return;
+    }
+    $this->entityTypeManager->getStorage('config_pages')->create([
+      'type' => 'stanford_basic_site_settings',
+      'su_site_email' => $site_data['email'],
+      'su_site_name' => $site_data['webSiteTitle'],
+      'context' => 'a:0:{}',
```

Comment:
What is this context?

Reply:
something in the config pages. i don't really know what its used for. So far as much as i can tell its for contextual changes with languages. but not sure beyond that. But basically the column in the database cant be null

Reply:
Thanks.